klippy/extras/safe_z_home.py

```diff
@@ -40,8 +40,8 @@ def cmd_G28(self, gcmd):
                 toolhead.set_position(pos, homing_axes=[2])
                 toolhead.manual_move([None, None, self.z_hop],
                                      self.z_hop_speed)
-                if hasattr(toolhead.get_kinematics(), "note_z_not_homed"):
-                    toolhead.get_kinematics().note_z_not_homed()
+                if hasattr(toolhead.get_kinematics(), "clear_homing_state"):
+                    toolhead.get_kinematics().clear_homing_state((2,))
```

Comment:
If we're making `clear_homing_state()` required for all kinematics, then I think it should be called unconditionally here (that is, no `hasattr()` check).